js/print.js

```diff
@@ -0,0 +1,316 @@
+    return link;
+  }
+
+  link.innerHTML = '<span class="secnum">' + clauseNumber + '</span> ' + clauseTitle;
+  return link;
+}
+
+/**
+ * Loops through every clause/annex's h1 and improves the markup
+ * */
+function improveSectionHeadings() {
+  // eslint-disable-next-line prettier/prettier
+  const sectionHeadings = Array.from(specContainer.querySelectorAll('emu-clause > h1, emu-annex > h1'));
+
+  /** these section IDs are emu-annex elements but not functionally annexes */
+  const nonAnnexSections = [
+    'sec-copyright-and-software-license',
+    'sec-colophon',
+    'sec-bibliography',
+  ];
+
+  sectionHeadings.forEach(h1 => {
+    const secnum = h1.firstChild;
+    const section = secnum.innerHTML;
+    const parent = h1.parentNode;
+
+    if (/^[A-Z]$/.test(section)) {
+      h1.classList.add('annex-title');
+
+      if (nonAnnexSections.includes(parent.id)) {
+        secnum.innerHTML = '';
+      } else {
+        const annexType = parent.hasAttribute('normative') ? 'normative' : 'informative';
+
+        secnum.innerHTML =
+          'Annex ' + section + ' <br/><span class="unbold">(' + annexType + ')</span><br/>';
+      }
+    } else {
+      secnum.textContent = section;
+    }
+
+    if (secnum.textContent !== '') {
+      h1.insertBefore(document.createTextNode(' '), h1.firstChild);
+      h1.insertBefore(secnum, h1.firstChild);
+    }
+  });
+}
+
+/**
+ * The emu-imports element interferes with a ton of css specificity so we just
+ * take everything out and plop it directly into the DOM
+ * */
+function removeEmuImports() {
+  const emuImports = Array.from(specContainer.getElementsByTagName('emu-import'));
+
+  emuImports.forEach(importedMarkup => {
+    while (importedMarkup.hasChildNodes()) {
+      importedMarkup.parentNode.insertBefore(importedMarkup.childNodes[0], importedMarkup);
+    }
+    importedMarkup.parentNode.removeChild(importedMarkup);
+  });
+}
+
+/**
+ * Sets up table captions and figcaptions for tables, which provides for
+ * continuation table captions.
+ * */
+function rearrangeTables() {
+  const tables = Array.from(document.getElementsByTagName('emu-table'));
+
+  tables.forEach(emuTable => {
+    const figcaption = emuTable.getElementsByTagName('figcaption')[0];
+    const tableCaptionText = figcaption.innerHTML;
+    const table = emuTable.getElementsByTagName('table')[0];
+    const captionElement = document.createElement('caption');
+
+    captionElement.innerHTML = tableCaptionText;
+
+    table.insertBefore(captionElement, table.getElementsByTagName('thead')[0]);
+    table.appendChild(figcaption);
+  });
+}
+
+/**
+ * Gets rid of elements we don't need in the print version
+ * */
+function removeUnusedSections() {
+  const ecmaLogo = document.getElementById('ecma-logo');
+
+  if (ecmaLogo) specContainer.removeChild(ecmaLogo.parentNode);
+
+  document.getElementsByTagName('body')[0].removeChild(document.getElementById('shortcuts-help'));
+}
```

Comment:
Surely this can be done with print CSS.

Reply:
of course it can. getting the scripts to output what i need is a finicky thing though and I try to feed it the simplest, cleanest markup i can.